expertsystem/io/__init__.py

```diff
@@ -0,0 +1,39 @@
+"""Serialization module for containers of `expertsystem.data`."""
```

Comment:
Just to make sure I understand this. Later on in the refactored expertsystem, this whole `io` package can be reduced to merely the load and dump functions, right?

Reply:
Yes I think this `io` module will eventually have only two, maybe three, wrapper functions:
* `write`
* `load`

I say three, because the `load` would have to take its interpretation from the recipe file. Then it would produce a `ParticleCollection` instance if the recipe file only contains a `"ParticleList"` section and a `Recipe` instance if the recipe file contains more than that. But to be honest, I think that's a bit to much implicit reasoning, so I would prefer a destinction here: `load_particle_collection` (as introduced in this PR) and `load_recipe` (follow-up PRs).

Note that `write` can get all the information from its input with `isinstance` checks, so one `write` function is sufficient (it takes it's output format from the file extension.

However, even on the surface, this module only offers two/three functions, there are a lot more implementation details underneath, so I would still keep this `io` module separate.